MSAL/src/requests/MSALInteractiveRequest.m

```diff
@@ -176,77 +105,98 @@ - (void)acquireToken:(MSALCompletionBlock)completionBlock
-     }];
-
+    void (^webAuthCompletion)(MSIDWebviewResponse *, NSError *) = ^void(MSIDWebviewResponse *response, NSError *error)
+    {
+        if ([response isKindOfClass:MSIDWebAADAuthResponse.class])
+        {
+            MSIDWebAADAuthResponse *oauthResponse = (MSIDWebAADAuthResponse *)response;
+            
+            if (oauthResponse.authorizationCode)
+            {
+                _code = oauthResponse.authorizationCode;
+                _cloudAuthority = [NSURL URLWithString:((MSIDWebAADAuthResponse *)response).cloudHostName];
+                
+                [super acquireToken:completionBlock];
+                return;
+            }
+            
+            completionBlock(nil, oauthResponse.oauthError);
+            return;
+        }
+        
+        else if([response isKindOfClass:MSIDWebMSAuthResponse.class])
+        {
+            // Todo: Install broker prompt
+        }
+        
+        completionBlock(nil, error);
+    };
     
+    switch (_parameters.webviewType) {
+        
+#if TARGET_OS_IPHONE
+        case MSALWebviewTypeAuthenticationSession:
+            [MSIDWebviewAuthorization startSystemWebviewAuthWithConfiguration:config
+                                                                oauth2Factory:_parameters.msidOAuthFactory
+                                                     useAuthenticationSession:YES
+                                                    allowSafariViewController:NO
+                                                                      context:_parameters
+                                                            completionHandler:webAuthCompletion];
+            break;
+
+        case MSALWebviewTypeSafariViewController:
+            [MSIDWebviewAuthorization startSystemWebviewAuthWithConfiguration:config
+                                                                oauth2Factory:_parameters.msidOAuthFactory
+                                                     useAuthenticationSession:NO
+                                                    allowSafariViewController:YES
+                                                                      context:_parameters
+                                                            completionHandler:webAuthCompletion];
+            break;
+        case MSALWebviewTypeAutomatic:
+            [MSIDWebviewAuthorization startSystemWebviewAuthWithConfiguration:config
+                                                                oauth2Factory:_parameters.msidOAuthFactory
+                                                     useAuthenticationSession:YES
+                                                    allowSafariViewController:YES
+                                                                      context:_parameters
+                                                            completionHandler:webAuthCompletion];
+            break;
+#else
+        case MSALWebviewTypeAutomatic:
```

Comment:
this is a bit difficult to read... not sure how to improve it though. Can we at least add a comment?